src/gui/sharedialog.cpp

```diff
@@ -161,36 +161,35 @@ void ShareDialog::addLinkShareWidget(const QSharedPointer<LinkShare> &linkShare)
     connect(linkShare.data(), &Share::shareDeleted, _linkWidgetList.at(index), &ShareLinkWidget::slotDeleteShareFetched);
 
     if(_manager) {
-        connect(_manager, &ShareManager::linkShareRequiresPassword, _linkWidgetList.at(index), &ShareLinkWidget::slotCreateShareRequiresPassword);
         connect(_manager, &ShareManager::serverError, _linkWidgetList.at(index), &ShareLinkWidget::slotServerError);
     }
 
     // Connect all shares signals to gui slots
     connect(this, &ShareDialog::toggleAnimation, _linkWidgetList.at(index), &ShareLinkWidget::slotToggleAnimation);
     connect(_linkWidgetList.at(index), &ShareLinkWidget::createLinkShare, this, &ShareDialog::slotCreateLinkShare);
     connect(_linkWidgetList.at(index), &ShareLinkWidget::deleteLinkShare, this, &ShareDialog::slotDeleteShare);
+    connect(_linkWidgetList.at(index), &ShareLinkWidget::createPassword, this, &ShareDialog::slotCreatePasswordForLinkShare, Qt::UniqueConnection);
```

Comment:
Are we sure `Qt::UniqueConnection` is really needed here? It's best avoided when possible, it has its uses but they better be the exception not the rule (relying on it is often a sign of slightly broken design, I treat it like a code smell).

I see it's not needed for the other ones so if we'd go through that bunch of connects more than once we'd be in trouble today anyway.